client/signup/steps/site-info-collection/index.tsx

```diff
@@ -0,0 +1,277 @@
+
+	// Initialize local state with the values from the redux store
+	const [ formValues, setFormValues ] = useState< SiteInfoCollectionData >(
+		siteInfoCollectionDataFromStore
+	);
+	const onChangeField = ( { target: { name, value } }: ChangeEvent< HTMLInputElement > ) => {
+		setFormValues( {
+			...formValues,
+			[ name ]: value,
+		} );
+	};
+
+	const [ formErrors, setFormErrors ] = useState< ValidationErrors >( {} );
+
+	const sections: SectionProps[] = [
+		{
+			title: translate( '%d. Name of your business', {
+				args: [ 1 ],
+				comment: 'This is the serial number: 1',
+			} ),
+			component: (
+				<TextInputField
+					label={ translate(
+						'Please enter the name of your business as you want it to appear on your new site.'
+					) }
+					value={ formValues.siteTitle }
+					error={ formErrors.siteTitle }
+					name="siteTitle"
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: false,
+			summary: formValues.siteTitle,
+			validate: ( { siteTitle } ) => {
+				const isValid = Boolean( siteTitle && siteTitle.length );
+				return {
+					result: isValid,
+					errors: {
+						siteTitle: isValid ? null : translate( 'Please enter a valid site title.' ),
+					},
+				};
+			},
+		},
+		{
+			title: translate( '%d. Site Description', {
+				args: [ 2 ],
+				comment: 'This is the serial number: 2',
+			} ),
+			component: (
+				<TextInputField
+					label={ translate( 'A short description for your website.' ) }
+					value={ formValues.siteDescription }
+					name="siteDescription"
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: formValues.siteDescription,
+		},
+		{
+			title: translate( '%d. Social Media Profiles', {
+				args: [ 3 ],
+				comment: 'This is the serial number: 3',
+			} ),
+			component: (
+				<SocialMediaProfiles
+					facebookProps={ {
+						value: formValues.facebookUrl,
+						name: 'facebookUrl',
+					} }
+					twitterProps={ {
+						value: formValues.twitterUrl,
+						name: 'twitterUrl',
+					} }
+					linkedinProps={ {
+						value: formValues.linkedinUrl,
+						name: 'linkedinUrl',
+					} }
+					instagramProps={ {
+						value: formValues.instagramUrl,
+						name: 'instagramUrl',
+					} }
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: undefined,
+		},
+		{
+			title: translate( '%d. Contact Information', {
+				args: [ 4 ],
+				comment: 'This is the serial number: 4',
+			} ),
+			component: (
+				<ContactInformation
+					displayEmailProps={ {
+						value: formValues.displayEmail,
+						name: 'displayEmail',
+					} }
+					displayPhoneProps={ {
+						value: formValues.displayPhone,
+						name: 'displayPhone',
+					} }
+					displayAddressProps={ {
+						value: formValues.displayAddress,
+						name: 'displayAddress',
+					} }
+					onChange={ onChangeField }
+				/>
+			),
+			showSkip: true,
+			summary: undefined,
+		},
+	];
+
+	const isSectionAtIndexTouchedInitialState: Record< string, boolean > = {};
```

Comment:
This needs to be at the parent level since the user can jump from any section to any other section. For example, if the currentIndex is 0, and the user clicks on the "Contact Information" section, the currentIndex will be 3. However, we cannot mark sections at index 1 & 2 as touched since they have not been seen by the user.